### PR TITLE
Merge two SP migration docs into a single file

### PR DESCRIPTION
**Pull request by ezgidemirel**
### Description of your changes
To eliminate duplicate steps, we're merging MR migration and Configuration migration guides into a single file. 

Fixes #

I have:

- [ ] Read and followed Crossplane's [contribution process].
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested
N/A

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: docs/sp-migration/sp-migration.md [hunk 1]
```diff
@@ -94,17 +93,13 @@ kubectl get managed
 kubectl get provider.pkg
 ```
 
-10. Find new dependencies and add them to `dependsOn` field in the `crossplane.yaml` file:
+10. If you want to add smaller providers to a configuration's `dependsOn` list, please follow the guide and build/push/update
```
Comment (turkenh) [inline docs/sp-migration/sp-migration.md:96]: `follow the guide`

Can we link that guide here?

**Review by turkenh — APPROVED** [review]

File: docs/sp-migration/sp-migration.md [hunk 2]
```diff
@@ -102,17 +110,18 @@ kubectl get managed
 kubectl get provider.pkg
 ```
 
-10. Find new dependencies and add them to `dependsOn` field in the `crossplane.yaml` file:
+10. If you want to add smaller providers to the configuration's `dependsOn` list, please follow the guide and build/push/update
```
Comment (sergenyalcin) [inline docs/sp-migration/sp-migration.md:113]: What is the referred guide here? We may consider giving a link.

**Review by sergenyalcin — COMMENTED** [review]
Thanks @ezgidemirel I left a comment

File: docs/sp-migration/sp-migration.md (hunk 2 above)
Comment (ezgidemirel) [reply docs/sp-migration/sp-migration.md]: Forgot to add parentheses, thanks for catching :)

**Pull request merged**